### DISCUSSION TITLE
Where do I find downloaded files?

### DISCUSSION THREAD

**Discussion by ShapeShifter499** [CappielloAntonio/tempo#182] [Q&A]
In the event my device is offline, where could I find the files I downloaded in case I wanted to share music in person?

This would include situations where internet is down, no signal, etc. But we have the correct cables to connect (otg or usb-c) two devices.

In Android 14+ I cannot access some folders in /Android on the root of the user folder. So if it's there maybe there could be a user selected download location for music?

**Comment by eddyizm** [comment]
It is probably in the `Android/data/<App-Name>` folder which is getting harder and harder to get to. I was using a 3rd party file explorer. 
if you enable developer debugging though, you can access it via `adb shell` althought I am not sure if these files are usable and they obviously aren't named the same. Being able to select a specific location besides internal/external so other apps can access the files would be great. 

'''
adb shell
r8q:/ $ ls storage/self/primary/Android/data/com.cappielloantonio.tempo/files/downloads/8
1.5242880.1719107490859.v3.exo     146.0.1719117389278.v3.exo         43.5242880.1719109677061.v3.exo
111.0.1719112671663.v3.exo         146.20971520.1719117401423.v3.exo  48.5242880.1719109689275.v3.exo
118.47185920.1719112771922.v3.exo  148.52428800.1719117419562.v3.exo  49.0.1719109687203.v3.exo
119.31457280.1719112771902.v3.exo  160.0.1719118803764.v3.exo         51.5242880.1719109691943.v3.exo
119.5242880.1719112754552.v3.exo   17.0.1719108298930.v3.exo          69.5242880.1719109719998.v3.exo
128.5242880.1719115885509.v3.exo   18.0.1719108298969.v3.exo          83.0.1719112598409.v3.exo
132.0.1719115887551.v3.exo         21.0.1719108302093.v3.exo          92.0.1719112604380.v3.exo
133.0.1719117290537.v3.exo         22.0.1719108302780.v3.exo          99.0.1719112607655.v3.exo
145.0.1719117376144.v3.exo         38.5242880.1719109089327.v3.exo
145.15728640.1719117397899.v3.exo  40.0.1719109089497.v3.exo
'''

**Reply by devinfriday** [reply]
If these are the music files downloaded from Tempo, it could be very hard that another music player could read those...

**Reply by eddyizm** [reply]
@devinfriday does seem nigh impossible.

**Comment by xxxserxxx** [comment]
This is more a question for @CappielloAntonio , but is there a reason why the app doesn't have a "select music folder" option? Sure, this is a Subsonic streamer, but local cache should come first, right? And if it accessed the Music directory, it could be used as a general purpose music app, as well as downloading files to a place _other_ music players and file browsers could find them.

**Reply by devinfriday** [reply]
Recent Android versions crippled the ability to choose a folder for your own. Simply you don't have the rights to. Period.

**Reply by xxxserxxx** [reply]
Is that true? I'm using a Samsung phone, so with Samsung's fork of Android, but it's based on Android 14. Is this restriction part of Android 15?

At any time I have three or four music apps I'm trying to decide between, and they all allow me to access my music. The Metro app (in f-droid) is what I'm using recently, because it lets me edit the song metadata tags.  Additionally, Tubular -- the NewPipe fork I prefer -- allows downloading YonTnbe music as audio files to a user-selected directory that you specify in the config, and I have that set to the `Music` directory.

**Comment by eddyizm** [comment]
@ShapeShifter499 @xxxserxxx  @devinfriday We've added the ability to select a user defined folder for downloads that allows access to the music file by the rest of the android os and other apps starting in v3.16.0 
-> https://github.com/eddyizm/tempo/releases
